sdk/iot/provisioning/tests/cmocka/test_az_iot_provisioning_client.c

```diff
@@ -75,60 +91,162 @@ static void test_az_iot_provisioning_client_custom_options_get_username_succeed(
       &options);
   assert_int_equal(AZ_OK, ret);
 
-  char user_name[128];
-  memset(user_name, 0xCC, sizeof(user_name));
-
-  char* expected_username = TEST_ID_SCOPE
+  char expected_username[] = TEST_ID_SCOPE
       "/registrations/" TEST_REGISTRATION_ID
       "/api-version=" AZ_IOT_PROVISIONING_SERVICE_VERSION USER_AGENT_PREFIX TEST_USER_AGENT;
 
-  ret = az_iot_provisioning_client_get_user_name(&client, user_name, sizeof(user_name), NULL);
+  char user_name[sizeof(expected_username) + 1];
+  memset(user_name, 0xCC, sizeof(user_name));
+  size_t user_name_len;
+  ret = az_iot_provisioning_client_get_user_name(
+      &client, user_name, sizeof(user_name), &user_name_len);
   assert_int_equal(AZ_OK, ret);
   assert_memory_equal(expected_username, user_name, strlen(user_name));
   assert_int_equal((uint8_t)0xCC, (uint8_t)user_name[strlen(user_name) + 1]);
+  assert_int_equal(strlen(expected_username), user_name_len);
 }
 
-static void test_az_iot_provisioning_client_get_subscribe_topic_filter_succeed()
+static void test_az_iot_provisioning_client_get_connect_info_insufficient_space_fails()
```

Comment:
What is connect info?

Reply:
MQTT CONNECT is one of the MQTT message types. The information that our client is computing in connect is `user_name`, `client_id` and `password`.